### DISCUSSION TITLE
Hide or Disable Form Fields based on other database fields?

### DISCUSSION THREAD

**Discussion by yaplej** [filamentphp/filament#2512] [Help] [answered]
Hello,

How can I disable a field based on the value of another field of the record or related record?  I have been able to figure out how to disable a field based on the Page being Edit vs Create from the [documentation](https://filamentphp.com/docs/2.x/admin/resources#hiding-components-based-on-the-page).

`->disabled(fn (Component $livewire): bool => $livewire instanceof Pages\EditRecord),`

I could not find an example of this in the documentation.

Thank you.

**Comment by danharrin** [comment, marked as answer]
This works if you want a saved model instance:

```php
->disabled(fn (Model $record) => $record->some_value),
```

or if you want the value from another field (make that field `reactive()`):

```php
->disabled(fn ($get) => $get('some_value')),
```

https://filamentphp.com/docs/2.x/forms/advanced#using-closure-customisation

**Comment by yaplej** [comment]
Thank you.  I was missing how to test for a specific value but it was pretty straight forward.
`->disabled(fn (Model $record) => $record->my_field == 'My Value'),`
